relay-metrics/src/aggregation.rs

```diff
@@ -1302,17 +1297,38 @@ impl Aggregator {
             return Err(AggregateMetricsErrorKind::InvalidStringLength.into());
         }
 
-        if !protocol::is_valid_mri(&key.metric_name) {
-            relay_log::debug!("invalid metric name {:?}", key.metric_name);
-            relay_log::configure_scope(|scope| {
-                scope.set_extra(
-                    "bucket.project_key",
-                    key.project_key.as_str().to_owned().into(),
-                );
-                scope.set_extra("bucket.metric_name", key.metric_name.into());
-            });
-            return Err(AggregateMetricsErrorKind::InvalidCharacters.into());
-        }
+        key.metric_name = match key.metric_name.parse::<MetricResourceIdentifier>() {
```

Comment:
_nit_: Since this is quite a block of code that normalizes the MRI, let's introduce a function for that.